src/QsCompiler/Compiler/CompilationLoader.cs

```diff
@@ -573,16 +580,17 @@ public CompilationLoader(SourceLoader loadSources, ReferenceLoader loadReference
             // executing the specified rewrite steps
             PerformanceTracking.TaskStart(PerformanceTracking.Task.RewriteSteps);
             var steps = new List<(int, string, Func<QsCompilation?>)>();
-            this.config.IsQirGenerationEnabled = this.externalRewriteSteps.Any(step => step.Name == "QIR Generation");
+            this.config.PrepareQirGeneration = this.config.PrepareQirGeneration || this.externalRewriteSteps.Any(step => step.Name == "QIR Generation");
 
-            if ((this.config.IsExecutable || this.config.IsQirGenerationEnabled) && !this.config.SkipSyntaxTreeTrimming)
+            if (this.config.IsExecutable && !this.config.SkipSyntaxTreeTrimming)
             {
                 // TODO: It would be nicer to trim unused intrinsics. Currently, this is not possible due to how the old setup of the C# runtime works.
                 // With the new setup (interface-based approach for target packages), it is possible to trim ununsed intrinsics.
 #pragma warning disable CS0618 // Type or member is obsolete
                 // TODO: The dependencies for rewrite steps should be declared as part of IRewriteStep interface, and we should query those here.
-                var rewriteStep = new LoadedStep(new SyntaxTreeTrimming(keepAllIntrinsics: !this.config.IsQirGenerationEnabled, BuiltIn.AllBuiltIns.Select(e => e.FullName), !this.config.IsExecutable), typeof(IRewriteStep), thisDllUri);
+                var trimming = new SyntaxTreeTrimming(keepAllIntrinsics: true, BuiltIn.AllBuiltIns.Select(e => e.FullName));

```

Comment:
This needs to be true! See the comment above:
"TODO: It would be nicer to trim unused intrinsics. Currently, this is not possible due to how the old setup of the C# runtime works."

Reply:
I unfortunately missed the conversation here at that time: https://github.com/microsoft/qsharp-compiler/pull/1062#discussion_r668125520

Reply:
Ah, I didn't realize that we already have scenarios that need to support C# generation and QIR generation simultaneously. I think that since QirGenration runs it's own syntax tree trimming step, it will still clean up the unused intrinsics, right?

Reply:
Yes, the QirGeneration runs its own syntax tree trimming.